functions/src/common.ts

```diff
@@ -8,13 +8,40 @@ export function safeString(unsafe: string):string {
   return unsafe.replace(/\//gi, "_");
 }
 
+export async function saveFields(jsonArray: any[], listId: string) {
+  let dict: any = {};
+  for (let item of jsonArray) {
+    for (let key in item) {
+      if (key in dict) {
+        dict[key]['count'] = dict[key]['count'] + 1;
+      } else {
+        dict[key] = {type: Array.isArray(item[key]) ? 'array' : typeof(item[key]), count: 1};
+      }
+    }
+  }
+  let colRef: CollectionReference = db.collection('list').doc(listId).collection('fields');
+  let counter = 0;
+  let batch = db.batch();
+  for (let key in dict) {
+    batch.set(colRef.doc(key), dict[key]);
+    counter++;
+    if (counter > FIRESTORE_WRITE_BATCH_SIZE) {
+      await batch.commit();
+      batch = db.batch();
+      counter = 0;
+    }
+  }
+  await batch.commit();
+}
+
 // Saving list
 export async function saveList(jsonArray: any[], listId: string, fieldId: string) {
   let docRef: DocumentReference = db.collection("list").doc(listId);
   let hash = createHash("md5").update(JSON.stringify(jsonArray)).digest("hex");
   console.dir(`fetched list document with hash: ${hash}`);
   let doc = await docRef.get();
   if (doc.exists && doc.data()?.["lastUpdateHash"] === hash) {
+    await saveFields(jsonArray, listId);
```

Comment:
Never do this in real life! Because we will forget to remove it and it will create a bug!
Just delete hash.